File: packages/StewardApi/StewardApi/Contracts/Common/LspTask.cs
using System;
using System.Collections.Generic;
using WoodstockLiveOpsContent;

#pragma warning disable CS1591
#pragma warning disable SA1600
namespace Turn10.LiveOps.StewardApi.Contracts.Common
{
    /// <summary>
    ///     Represents an LSP task.
    /// </summary>
    public sealed class LspTask
    {
        public Guid Id { get; set; }

        public string ExecutorType { get; set; }

        public DateTime LastEventUtc { get; set; }

        public string LastException { get; set; }

        public DateTime NextExecutionUtc { get; set; }

        public double PeriodInSeconds { get; set; }

        public LspTaskPeriodType PeriodType { get; set; }

        public double LastRunDuration { get; set; }

        public LspTaskState State { get; set; }

        public Guid Lock { get; set; }

        public DateTime LockTakenUntilUtc { get; set; }

        public string CustomProperties { get; set; }
    }
}

File: packages/StewardClient/src/app/shared/models/lsp-task.ts
import { DateTime } from 'luxon';
import { GuidLikeString } from './extended-types';
import BigNumber from 'bignumber.js';

export enum LspTaskState {
  Pending = 'Pending',
  Processing = 'Processing',
  Complete = 'Complete',
  Failed = 'Failed',
  Disabled = 'Disabled',
}

export enum LspTaskPeriodType {
  None = 'None',
  Deterministic = 'Deterministic',
  NonDeterministic = 'NonDeterministic',
}

/** Interface for a an lsp task. */
export interface LspTask {
  customProperties: string;
  executorType: string;
  id: GuidLikeString;
  lastEventUtc: DateTime;
  lastException: string;
  lastRunDuration: BigNumber;
  lock: GuidLikeString;
  lockTakenUntilUtc: DateTime;
  nextExecutionUtc: DateTime;
  periodInSeconds: BigNumber;
  periodType: LspTaskPeriodType;
  state: LspTaskState;
}
